08_project/mohankrishna/pkg/mohankrishna-puppy/store/leveldb_store.go

```diff
@@ -0,0 +1,85 @@
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	types "github.com/anz-bank/go-training/08_project/mohankrishna/pkg/mohankrishna-puppy"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+type LevelDBStore struct {
+	ldb   *leveldb.DB
+	mutex sync.Mutex
+}
+
+func NewLevelDBStore(dbPath string) *LevelDBStore {
+	db, _ := leveldb.OpenFile(dbPath, nil)
+	return &LevelDBStore{ldb: db}
+}
+
+func (ls *LevelDBStore) CloseDB() {
+	ls.ldb.Close()
+}
+
+func (ls *LevelDBStore) CreatePuppy(puppy *types.Puppy) error {
+	b, _ := json.Marshal(puppy.ID)
+	ls.mutex.Lock()
+	defer ls.mutex.Unlock()
+	ok, _ := ls.ldb.Has(b, nil)
+	if ok {
+		return &types.Error{Code: types.ErrDuplicate,
+			Message: fmt.Sprintf("A puppy with ID: %d already exists", puppy.ID)}
+	}
+	bytes, _ := json.Marshal(puppy)
+	err := ls.ldb.Put(b, bytes, nil)
+	return err
+}
+
+func (ls *LevelDBStore) ReadPuppy(id uint32) (*types.Puppy, error) {
+	b, _ := json.Marshal(id)
+	data, err := ls.ldb.Get(b, nil)
+	if err != nil {
+		return nil, err
+	}
+	var puppy types.Puppy
+	err = json.Unmarshal(data, &puppy)
+	return &puppy, err
+}
+
+func (ls *LevelDBStore) UpdatePuppy(id uint32, puppy *types.Puppy) error {
+	if id != puppy.ID {
+		return &types.Error{Code: types.ErrInvalidInput,
+			Message: fmt.Sprintf("The id:%d passed and the puppy's id:%d doesnt match", id, puppy.ID)}
+	}
+	b, _ := json.Marshal(id)
+	bytes, _ := json.Marshal(puppy)
```

Comment:
Handle errors. If they should be impossible for the given input, then `panic(err)` instead of `return err`.